### DISCUSSION TITLE
Can I use both Page Numbers and Section/Header Numbers?

### DISCUSSION THREAD

**Discussion by groundbirdaircat** [dolanmiu/docx#2471] [Q&A] [answered]
I am trying to implement a document that has Header/Section numbers, as well as overall page numbers.

My specific use case calls for Section numbers in the Header, and overall Page numbers in the Footer.

For example, on Page 2 of the document, if I start a new Section, I would want the Section Header to display Page 1, but in the Footer of the page, it would display Page 2.

It appears that both of these numbers are accessed by `PageNumber.CURRENT`, and that setting the `pageNumbers` property in the page settings overwrites this value for both, since it's actually the same property and there is no way to distinguish between current Page or Section numbers.

Just to be clear, you are able to achieve this for **total** page numbers, as you can use `PageNumber.TOTAL_PAGES` or `PageNumber.TOTAL_PAGES_IN_SECTION`, but for the **current** page, there is only `PageNumber.CURRENT`.

I expected to see something similar to `PageNumber.CURRENT_IN_SECTION`, but have had no luck in finding this.

Is there something that I am missing here, or are there any known workarounds for this?

Any info would be greatly appreciated. Thanks!

**Comment by dolanmiu** [comment, marked as answer]
I do not think this is possible actually after doing some research. Even in Microsoft Word, this is incredibly difficult to do with workarounds.

This is the workaround needed with Word: https://askalibrarian.csuohio.edu/faq/43427

You can imagine how complex `docx` would have to be

@groundbirdaircat

**Reply by groundbirdaircat** [reply]
@dolanmiu

Thank you for your reply, I appreciate it.

As one last hope for a workaround, I have been experimenting with using getters to try to track the page number manually. However, it seems that `docx` is grabbing a reference internally to the return of the `default` getter that I put in the headers property of the section, rather than calling the getter every time.. This results in repeated numbers, so it doesn't work.

So, forgive me if this is a naive question, but would there be an easy change that could cause default to be called every time when using a getter? This could allow us to track the section numbers and use a wrapper class to inject the number manually, rather than trying to do it under the hood.

Here's a simple example for clarity:

```ts
headers: {
  get default() {
    manualSectionTracking++;
    return new CustomHeader(manualSectionTracking).getHeader();
  }
}
```

**Reply by dolanmiu** [reply]
Hmm I don't fully understand the question, but probably not

that would only get called once:

https://github.com/dolanmiu/docx/blob/master/src/file/file.ts#L114-L132
